src/main/scala/ztis/model/RandomPrediction.scala

```diff
@@ -0,0 +1,33 @@
+package ztis.model
+
+import org.apache.spark.SparkContext._
+import org.apache.spark.mllib.recommendation.Rating
+import org.apache.spark.rdd.RDD
+
+import scala.util.Random
+
+class RandomPrediction(trainingData: RDD[Rating]) extends Predictor with Serializable {
```

Comment:
That is not the cause - the problem are lambdas that are closing over "this" reference - we can try to fix it later.